### DISCUSSION TITLE
Calculate PP value of a map in Song Select

### DISCUSSION THREAD

**Discussion by mmodules** [ppy/osu#15763] [Ideas] [answered]
I am aware a song select overhaul is coming in the future, so I'd like to leave this in the air in case there's room for an extra feature.

### Issue
Many players use third-party tools such as bots or web extensions to calculate the amount of pp a map awards with a certain combo, accuracy, and mod combination. Although most people have gotten used to this, it is quite a pain dealing with inaccurate values after reworks, or having to leave the client to paste the map into Discord every time you submit a play.

### Idea
To counter this, we could implement something similar in the client where the player can choose to see this information.
A way this could possibly be done is by adding a _'performance'_ tab next to the other tabs in song select; as shown in the image below. 
**_(if this layout is similar after the overhaul)_**

![performance_tab](https://user-images.githubusercontent.com/73619648/143016017-770771af-435c-4fe0-8e38-ea58b95c8030.png)

Sorry if this is a duplicate! I searched for quite a while and couldn't find a similar proposal, so I posted this.

**Comment by bdach** [comment]
if this happens it will not look like the above. the other buttons on the bottom are menus not tabs.

**Reply by mmodules** [reply]
If this can be added and there is a need for someone to create a menu/new interface I'd be more than happy to help.

**Comment by smoogipoo** [comment, marked as answer]
I don’t think we’ll have an area where you can simulate arbitrary scores and all that requiring a button + interface, because that’s too elitist imo. There are external tools to do that, including [osu-tools](https://github.com/ppy/osu-tools).

But, I think we can have a piece of text in the beatmap details area displaying pp for 95%/100% accuracy FC.
